README.md

```diff
@@ -1,58 +1,145 @@
+First, get ExternalDNS.
 
-## Nice to have
+```console
+$ go get -u github.com/kubernetes-incubator/external-dns
+```
 
-* Should do smart cloud provider updates, i.e. Cloud Provider API should be called only when necessary
-* High Availability - should be possible to run multiple instances of External DNS
-* Should be able to monitor resource changes via K8S API for quick updates
-* New DNS record sources (e.g. TPRs) and targets (e.g. Azure DNS) should be pluggable and easy to add
+Run an application and expose it via a Kubernetes Service.
 
-## Example
+```console
+$ kubectl run nginx --image=nginx --replicas=1 --port=80
+$ kubectl expose deployment nginx --port=80 --target-port=80 --type=LoadBalancer
+```
 
-The [tutorials](docs/tutorials/gke.md) section contains a detailed example of how to setup `external-dns` on Google Container Engine.
+Annotate the Service with your desired external DNS name. Make sure to change `example.org` to your domain and that it includes the trailing dot.
 
-## Building
+```console
+$ kubectl annotate service nginx "external-dns.alpha.kubernetes.io/hostname=nginx.example.org."
+```
 
-You need a working Go 1.7+ development environment. Then run `make build` to build `external-dns` for your platform. The binary will land at `build/external-dns`.
+Run a single sync loop of ExternalDNS locally. Make sure to change the Google project to one you control and the zone identifier to an **unused** and **empty** hosted zone in that project's Google CloudDNS.
 
-## Getting involved!
+```console
+$ external-dns --zone example-org --google-project example-project --once
+```
 
-Want to contribute to External DNS? We would love the extra help from the community.
+This should output the DNS records it's going to modify to match the managed zone with the DNS records you desire.
 
-Reach out to us on [Kubernetes slack](https://github.com/kubernetes/community#slack-chat).
+Once you're satisfied with the result you can run ExternalDNS like you would run it in your cluster: as a control loop and not in dryRun mode.
+
+```console
+$ external-dns --zone example-org --google-project example-project --dry-run=false
+```
+
+Check that ExternalDNS created the desired DNS record for your service and that it points to its load balancer's IP. Then try to resolve it.
+
+```console
+$ dig +short nginx.example.org.
+104.155.60.49
+```
+
+Now you can experiment and watch how ExternalDNS makes sure that your DNS records are configured as desired. Here are a couple of things you can try out:
+* Change the desired hostname by modifying the Service's annotation.
+* Recreate the Service and see that the DNS record will be updated to point to the new load balancer IP.
+* Add another Service to create more DNS records.
+* Remove Services to clean up your managed zone.
+
+The [tutorials](docs/tutorials/gke.md) section contains a more detailed example of how to setup ExternalDNS in your Google Container Engine cluster.
```

Comment:
That makes sense. What do you think about having the rough workflow in the main readme at this point in geneal? Personally, I would like to have it here because it may attract a couple more people (devs) instead of on some sub page. I'm totally fine with naming it something else if that makes sense.